base/iterators.jl

```diff
@@ -1700,6 +1701,10 @@ function _nth(::IteratorSize, itr, n::Integer)
     y === nothing && throw(BoundsError(itr, n))
     y[1]
 end
+
+# issue #58922
```

Comment:
```suggestion
```

this specialization is reasonable to add no matter the motivating issue :)

Reply:
Resolved in 88639c0 (rebased to keep history clean)